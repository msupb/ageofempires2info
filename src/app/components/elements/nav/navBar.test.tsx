import React from 'react';
import ReactDom from 'react-dom';
import NavBar from './navBar';
import { BrowserRouter as Router } from 'react-router-dom';
import { cleanup, render } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';

afterEach(cleanup);

test("NavBar component renders properly", () => {
    const div = document.createElement('div');
    ReactDom.render(<Router><NavBar navClass="nav" linkClass="test" paths={['test', 'test2']}></NavBar></Router>, div);
    ReactDom.unmountComponentAtNode(div);
})

it("renders ul", () => {
    const { getByTestId } = render(<Router><NavBar navClass="nav" linkClass="test" paths={['test']}></NavBar></Router>);
    expect(getByTestId('navul')).toHaveTextContent('TEST');
})

it("matches snapshot", () => {
    const jsonObj = TestRenderer.create(<Router><NavBar navClass="nav" linkClass="test" paths={['test']}></NavBar></Router>).toJSON();
    expect(jsonObj).toMatchSnapshot();
})