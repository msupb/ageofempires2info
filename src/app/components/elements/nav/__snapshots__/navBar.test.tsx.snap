// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sidebar-container"
  style={
    Object {
      "backgroundImage": "url(/images/image3.jpg)",
      "backgroundSize": "cover",
    }
  }
>
  <div
    className="sidebar-logo"
  >
    Age of Empires 2 info
  </div>
  <ul
    className="sidebar-navigation"
    data-testid="navul"
  >
    <a
      className="test"
      href="/test"
      onClick={[Function]}
    >
      TEST
    </a>
  </ul>
</div>
`;
